common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,634 @@
+    }
+
+    @SubscribeEvent(priority = EventPriority.LOW)
+    public void onItemTooltipRenderEvent(ItemTooltipRenderEvent.Pre event) {
+        if (!KeyboardUtils.isKeyDown(holdToCompareKeyBind.getKeyMapping().key.getValue())) return;
         if (McUtils.mc().screen == null
                 || !(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
 
-        if (abstractContainerScreen.hoveredSlot == null) {
-            return;
-        }
+        if (abstractContainerScreen.hoveredSlot == null) return;
 
         ItemStack hoveredItemStack = abstractContainerScreen.hoveredSlot.getItem();
 
-        if (event.getItemStack() != hoveredItemStack) {
-            return;
-        }
+        if (event.getItemStack() != hoveredItemStack) return;
 
-        Optional<GearTypeItemProperty> gearItemOpt =
-                Models.Item.asWynnItemProperty(hoveredItemStack, GearTypeItemProperty.class);
-        if (gearItemOpt.isEmpty()) return;
+        Optional<WynnItem> hoveredWynnItemOpt = Models.Item.getWynnItem(hoveredItemStack);
+        if (hoveredWynnItemOpt.isEmpty()) return;
+        WynnItem hoveredWynnItem = hoveredWynnItemOpt.get();
 
-        ItemStack itemToCompare = null;
+        if (!(hoveredWynnItem instanceof GearTypeItemProperty hoveredGearItemProperty)) return;
 
-        // No compared item selected, try compare to equipped armor
-        if (compareToEquipped) {
-            List<ItemStack> armorSlots = McUtils.inventory().armor;
+        List<Pair<WynnItem, ItemStack>> itemsToCompare = new ArrayList<>();
 
-            Optional<ItemStack> matchingArmorItemStack = armorSlots.stream()
-                    .filter(itemStack -> isMatchingType(itemStack, gearItemOpt.get()))
-                    .findFirst();
+        // If hovered over selected item, only compare with other selected items
+        if (!selectedItems.containsKey(hoveredWynnItem)) {
+            switch (hoveredGearItemProperty.getGearType()) {
+                case HELMET, CHESTPLATE, LEGGINGS, BOOTS -> {
+                    List<ItemStack> armors = McUtils.inventory().armor;
 
-            itemToCompare = matchingArmorItemStack.orElse(null);
-        } else if (comparedItem != null) {
-            itemToCompare = comparedItem;
+                    Optional<ItemStack> matchingArmorOpt = armors.stream()
+                            .filter(itemStack -> isMatchingType(itemStack, hoveredGearItemProperty))
+                            .findFirst();
+                    if (matchingArmorOpt.isPresent()) {
+                        ItemStack armor = matchingArmorOpt.get();
+                        Models.Item.getWynnItem(armor).ifPresent(wynnItem -> {
+                            if (armor != hoveredItemStack) {
+                                itemsToCompare.add(new Pair<>(wynnItem, armor));
+                                ++equippedCount;
```

Comment:
A style issue: Please use postfix increment, unless required by the context to read the value (which, btw, is discouraged). This applies to all these  ++equippedCount lines.